feat: accept `Uint8Array`s as inputs for `Vec`s, add docs for bytecode inputs (#2018)
FuelLabs/fuels-ts: apps/docs/src/guide/types/vectors.md

<<< @/../../docs-snippets/src/guide/types/vector.test.ts#vector-4{ts:line-numbers}

## Working with Bytecode in the SDK

Some Sway functions require you to pass in bytecode to the function. The type of the bytecode parameter is usually `Vec<u8>`.

Take the `compute_bytecode_root` function from the [`bytecode` Sway library](https://github.com/FuelLabs/sway-libs/tree/master/libs/src/bytecode.sw), for example.

<<< @/../../docs-snippets/test/fixtures/forc-projects/bytecode-input/src/main.sw#vector-bytecode-input-sway{ts:line-numbers}

> To pass in some bytecode to this function, you can make use of the `arrayify` function to convert the bytecode file contents into a `Vec<u8>` type and pass it to the function like this:

Reviewer: ```suggestion
To pass bytecode to this function, you can make use of the `arrayify` function to convert the bytecode file contents into a `UInt8Array`, the TS compatible type for Sway's `Vec<u8>` type and pass it the function like so:
```

Just amended to show the TS -> Sway flow.
